engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpression.scala

```diff
@@ -138,10 +161,19 @@ class SpelExpressionParser(parser: org.springframework.expression.spel.standard.
   }
 
   override def parse(original: String, ctx: ValidationContext, expectedType: TypingResult): Validated[NonEmptyList[ExpressionParseError], TypedExpression] = {
-    baseParse(original).andThen { parsed =>
-      validator.validate(parsed, ctx, expectedType).map((_, parsed))
-    }.map { case (combinedResult, parsed) =>
-      TypedExpression(expression(ParsedSpelExpression(original, () => baseParse(original), parsed), expectedType), combinedResult.finalResult, combinedResult.typingInfo)
+    if (StringUtils.isBlank(original)) {
+      val typingResult = CollectedTypingResult.withEmptyIntermediateResults(expectedType)
+      Valid(TypedExpression(
+        NullExpression(original, expectedType, flavour),
+        typingResult.finalResult,
```

Comment:
can u use here directly `expectedType` and it 3rd param just `SpelExpressionTypingInfo(Map.empty)`. Currently it is tricky that you create instance just to extract it one line below :)

Reply:
Ok, done